modules/tournament/src/main/arena/ScoringSystem.scala

```diff
@@ -53,29 +57,32 @@ private[tournament] object ScoringSystem extends AbstractScoringSystem {
         }
         else NoBerserk
         (p.winner match {
-          case None if p.quickDraw => Score(Some(false), Normal, berserk)
+          case None if p.quickDraw => Score(ResDQ, Normal, berserk)
           case None => Score(
-            None,
+            ResDraw,
             if (isOnFire(scores)) Double else Normal,
             berserk
           )
           case Some(w) if userId == w => Score(
-            Some(true),
+            ResWin,
             if (isOnFire(scores)) Double
             else if (scores.headOption ?? (_.flag == StreakStarter)) StreakStarter
-            else n.flatMap(_.winner) match {
-              case Some(w) if userId == w => StreakStarter
+            else n match {
+              case None => StreakStarter
+              case Some(s) if s.winner.contains(userId) => StreakStarter
               case _ => Normal
             },
             berserk
           )
-          case _ => Score(Some(false), Normal, berserk)
+          case _ => Score(ResLoss, Normal, berserk)
         }) :: scores
     }
   }
 
   private def isOnFire = firstTwoAreWins _
 
-  private def firstTwoAreWins(scores: List[Score]) =
-    (scores.size >= 2) && (scores take 2 forall (~_.win))
+  private def firstTwoAreWins(scores: List[Score]) = scores match {
+    case List(Score(ResWin, _, _), Score(ResWin, _, _), _*) => true
```

Comment:
Yes, this works for any size >= 2.  However I think the cons matcher is clearer, so I've updated this code.